inference/core/workflows/core_steps/common/utils.py

```diff
@@ -48,38 +50,72 @@ def attach_prediction_type_info(
     return predictions
 
 
-def attach_parent_info(
-    images: List[Dict[str, Any]],
+def convert_to_sv_detections(
     predictions: List[Dict[str, Any]],
-    nested_key: Optional[str] = "predictions",
-) -> List[Dict[str, Any]]:
-    return [
-        attach_parent_info_to_prediction(
-            image=image, prediction=prediction, nested_key=nested_key
-        )
-        for image, prediction in zip(images, predictions)
-    ]
+    predictions_key: str = "predictions",
+    keypoints_key: str = "keypoints",
+    detection_id_key: str = "detection_id",
+    parent_id_key: str = PARENT_ID_KEY,
+) -> List[Dict[str, Union[sv.Detections, Any]]]:
+    converted_predictions: List[Dict[str, Union[sv.Detections, Any]]] = []
+    for p in predictions:
+        converted_prediction = deepcopy(p)
+        detections = sv.Detections.from_inference(p)
+        if any(keypoints_key in d for d in p[predictions_key]):
+            # keypoints arrays may have different length for each detection hence "object" type is used
+            detections[keypoints_key] = np.array(
+                [
+                    (
+                        np.array(
+                            [
+                                [keypoint["x"], keypoint["y"]]
+                                for keypoint in d[keypoints_key]
+                            ]
+                        )
+                        if d.get(keypoints_key)
+                        else np.array([])
+                    )
+                    for d in p[predictions_key]
+                ],
+                dtype="object",
+            )
+        if any(parent_id_key in d for d in p[predictions_key]):
+            detections[parent_id_key] = np.array(
+                [
+                    d[parent_id_key] if d.get(parent_id_key) else None
+                    for d in p[predictions_key]
+                ]
+            )
+        detection_ids = [
+            d[detection_id_key] if detection_id_key in d else str(uuid.uuid4)
+            for d in p[predictions_key]
+        ]
+        detections[detection_id_key] = np.array(detection_ids)
+        converted_prediction[predictions_key] = detections
+        converted_predictions.append(converted_prediction)
+    return converted_predictions
 
 
-def attach_parent_info_to_prediction(
-    image: Dict[str, Any],
-    prediction: Dict[str, Any],
-    nested_key: Optional[str],
-) -> Dict[str, Any]:
-    prediction[PARENT_ID_KEY] = image[PARENT_ID_KEY]
-    if nested_key is None:
-        return prediction
-    for detection in prediction[nested_key]:
-        detection[PARENT_ID_KEY] = image[PARENT_ID_KEY]
-    return prediction
+def attach_parent_info(
```

Comment:
[requires change] `nested_key` probably should be no longer needed and this function only requires to be used by blocks that do not operate on `sv.Detections` data format

Reply:
This method looked like this before:

```python
def attach_parent_info_to_prediction(
    image: Dict[str, Any],
    prediction: Dict[str, Any],
    nested_key: Optional[str],
) -> Dict[str, Any]:
    prediction[PARENT_ID_KEY] = image[PARENT_ID_KEY]
    if nested_key is None:
        return prediction
    for detection in prediction[nested_key]:
        detection[PARENT_ID_KEY] = image[PARENT_ID_KEY]
    return prediction
 ```

So originally `PARENT_ID_KEY` was added both as part of output and also as property for each prediction. I can drop nested_key and I can change the behavior of this method so parent ID is not added to each detection but it will change how this method was working.

Reply:
I would keep it for each prediction now